ui/settings.ts

```diff
@@ -0,0 +1,211 @@
+        settings.historical_data_start = $('#historical_data_start').val().toString();
+        let main_currency = $('#maincurrencyselector').val();
+        for (let i = 0; i < settings.CURRENCIES.length; i++) {
+            if (main_currency === settings.CURRENCIES[i].ticker_symbol) {
+                settings.main_currency = settings.CURRENCIES[i];
+            }
+        }
+
+        let eth_rpc_port = $('#eth_rpc_port').val();
+        let balance_save_frequency = $('#balance_save_frequency').val();
+        let send_payload = {
+            'ui_floating_precision': settings.floating_precision,
+            'historical_data_start': settings.historical_data_start,
+            'main_currency': main_currency,
+            'eth_rpc_port': eth_rpc_port,
+            'balance_save_frequency': balance_save_frequency
+        };
+        // and now send the data to the python process
+        client.invoke(
+            'set_settings',
+            send_payload,
+            (error, res) => {
+                if (error || res == null) {
+                    showError('Settings Error', 'Error at modifying settings: ' + error);
+                    return;
+                }
+                if (!res.result) {
+                    showError('Settings Error', 'Error at modifying settings: ' + res.message);
+                    return;
+                }
+
+                let message = 'Succesfully modified settings.';
+                if ('message' in res && res.message !== '') {
+                    message = ' ' + message + res.message;
+                }
+                showInfo('Success', message);
+        });
+    });
+
+    $('#historical_data_start').datetimepicker({timepicker: false});
+}
+
+export function create_settings_ui() {
+    let str = page_header('Settings');
+    str += settings_panel('General Settings', 'general_settings');
+    $('#page-wrapper').html(str);
+
+    str = form_entry(
+        'Floating Precision', 
+        'floating_precision', 
+        settings.floating_precision, ''
+    );
+    str += form_entry(
+        'Date from when to count historical data', 
+        'historical_data_start', 
+        settings.historical_data_start, ''
+    );
+    str += form_select(
+        'Select Main Currency', 'maincurrencyselector', 
+        settings.CURRENCIES.map(x => x.ticker_symbol), settings.main_currency.ticker_symbol
+    );
+    str += form_entry(
+        'Eth RPC Port', 
+        'eth_rpc_port', 
+        settings.eth_rpc_port, 
+        ''
+    );
+    str += form_entry(
+        'Balance data saving frequency in hours',
+        'balance_save_frequency',
+        settings.balance_save_frequency,
+        ''
+    );
+    $(str).appendTo($('.panel-body'));
+
+    str = form_button('Save', 'settingssubmit');
+    str += '</form></div>';
+    $(str).appendTo($('.panel-body'));
+}
+
+interface Settings {
+    usd_to_fiat_exchange_rates: any;
```

Comment:
This is an object with exchange rates. i.e.: `{'EUR': 1.232, 'CNY': 200.2}` e.t.c.